hydromt/components/mesh.py

```diff
@@ -12,33 +12,376 @@
+
+    def create(
+        self,
+        region: dict,
+        res: Optional[float] = None,
+        crs: Optional[int] = None,
+        grid_name: str = "mesh2d",
+    ) -> xu.UgridDataset:
+        """HYDROMT CORE METHOD: Create an 2D unstructured mesh or reads an existing 2D mesh according UGRID conventions.
+
+        Grids are read according to UGRID conventions. An 2D unstructured mesh
+        will be created as 2D rectangular grid from a geometry (geom_fn) or bbox.
+        If an existing 2D mesh is given, then no new mesh will be generated but an extent
+        can be extracted using the `bounds` argument of region.
+
+        Note Only existing meshed with only 2D grid can be read.
+
+        Adds/Updates model layers:
+
+        * **grid_name** mesh topology: add grid_name 2D topology to mesh object
+
+        Parameters
+        ----------
+        region : dict
+            Dictionary describing region of interest, bounds can be provided for type 'mesh'.
+            In case of 'mesh', if the file includes several grids, the specific 2D grid can
+            be selected using the 'grid_name' argument.
+            CRS for 'bbox' and 'bounds' should be 4326; e.g.:
+
+            * {'bbox': [xmin, ymin, xmax, ymax]}
+
+            * {'geom': 'path/to/polygon_geometry'}
+
+            * {'mesh': 'path/to/2dmesh_file'}
+
+            * {'mesh': 'path/to/mesh_file', 'grid_name': 'mesh2d', 'bounds': [xmin, ymin, xmax, ymax]}
+        res: float
+            Resolution used to generate 2D mesh [unit of the CRS], required if region
+            is not based on 'mesh'.
+        crs : EPSG code, int, optional
+            Optional EPSG code of the model or "utm" to let hydromt find the closest projected CRS.
+            If None using the one from region, and else 4326.
+        grid_name : str, optional
+            Name of the 2D grid in mesh, by default "mesh2d".
+
+        Returns
+        -------
+        mesh2d : xu.UgridDataset
+            Generated mesh2d.
+
+        """  # noqa: E501
+        self._logger.info("Preparing 2D mesh.")
+
+        # Create mesh2d
+        mesh2d = create_mesh2d(
+            region=region,
+            res=res,
+            crs=crs,
+            logger=self._logger,
+        )
+        # Add mesh2d to self.mesh
+        self.set(mesh2d, grid_name=grid_name)
+
+        # This setup method returns mesh2d so that it can be wrapped for models
+        # which require more information
+        return mesh2d
+
+    @property
+    def data(self) -> Union[xu.UgridDataArray, xu.UgridDataset]:
+        """
+        Model static mesh data. It returns a xugrid.UgridDataset.
+
+        Mesh can contain several grids (1D, 2D, 3D) defined according
+        to UGRID conventions. To extract a specific grid, use get_mesh
+        method.
+        """
+        # XU grid data type Xarray dataset with xu sampling.
+        if self._data is None and self._root.is_reading_mode():
+            self.read()
```

Comment:
I miss this `_initialize_mesh` function that we also see in the grid component. Would be nice to have it consistent.